src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -1203,6 +1207,50 @@ void makeSnapshotFromSnapshotRequestByRequestId() {
     assertThat(actual.getCreationInformation(), is(contentsModel));
   }
 
+  @Test
+  void getSourceDatasetsFromSnapshotRequestHandlesByRequestId() {
+    UUID snapshotAccessRequestId = UUID.randomUUID();
+    SnapshotRequestModel snapshotRequestModel = new SnapshotRequestModel();
+    SnapshotRequestContentsModel contentsModel = new SnapshotRequestContentsModel();
+    contentsModel.datasetName("datasetName");
+    contentsModel.mode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID);
+    SnapshotRequestIdModel requestIdModel = new SnapshotRequestIdModel();
+    requestIdModel.snapshotRequestId(snapshotAccessRequestId);
+    contentsModel.requestIdSpec(requestIdModel);
+    snapshotRequestModel.contents(List.of(contentsModel));
+
+    SnapshotAccessRequestResponse snapshotAccessRequest =
+        new SnapshotAccessRequestResponse().sourceSnapshotId(snapshotId);
+
+    Snapshot snapshot =
+        new Snapshot()
+            .snapshotSources(List.of(new SnapshotSource().dataset(new Dataset().id(datasetId))));
+
+    when(snapshotRequestDao.getById(snapshotAccessRequestId)).thenReturn(snapshotAccessRequest);
+    when(snapshotDao.retrieveSnapshot(snapshotId)).thenReturn(snapshot);
+
+    List<Dataset> datasets = service.getSourceDatasetsFromSnapshotRequest(snapshotRequestModel);
+
+    assertThat(datasets.get(0).getId(), is(datasetId));
```

Comment:
This could assert the dataset itself
```suggestion
    assertThat(datasets.get(0), is(snapshot.getSourceDataset()));
```